runtime/datetime.cpp

```diff
@@ -372,18 +367,24 @@ string f$date(const string &format, int64_t timestamp) {
 }
 
 bool f$date_default_timezone_set(const string &s) {
-  if (s != string("Etc/GMT-3", 9) && s != string("Europe/Moscow", 13)) {//TODO
-    if (s == string("Etc/GMT-4", 9)) {
-      php_warning("Timezone Etc/GMT-4 is not supported, use Etc/GMT-3 instead");
-      return true;
-    }
-    php_critical_error ("unsupported default timezone \"%s\"", s.c_str());
+  if (strcmp(s.c_str(), PHP_TIMELIB_TZ_GMT3) == 0) {
+    set_default_timezone_id(PHP_TIMELIB_TZ_GMT3);
+    return true;
+  }
+  if (strcmp(s.c_str(), PHP_TIMELIB_TZ_MOSCOW) == 0) {
+    set_default_timezone_id(PHP_TIMELIB_TZ_MOSCOW);
+    return true;
+  }
+  // TODO: this branch is weird; remove it?
+  if (strcmp(s.c_str(), PHP_TIMELIB_TZ_GMT4) == 0) {
+    php_warning("Timezone Etc/GMT-4 is not supported, use Etc/GMT-3 instead");
```

Comment:
may be
`php_warning("Timezone %s is not supported, use %s instead", PHP_TIMELIB_TZ_GMT4, PHP_TIMELIB_TZ_GMT3);`
?

Reply:
Done.